runtime/vm/jfr.cpp

```diff
@@ -729,6 +747,91 @@ jfrExecutionSampleCallback(J9VMThread *currentThread, IDATA handlerKey, void *us
 	jfrExecutionSample(currentThread, currentThread);
 }
 
+void
+jfrCPULoad(J9VMThread *currentThread)
+{
+	PORT_ACCESS_FROM_VMC(currentThread);
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+
+	omrthread_process_time_t currentProcCPUTime = {0};
+	intptr_t processTimeRC = omrthread_get_process_times(&currentProcCPUTime);
+
+	J9SysinfoCPUTime currentSysCPUTime = {0};
+	intptr_t sysTimeRC = omrsysinfo_get_CPU_utilization(&currentSysCPUTime);
+
+	if (0 == processTimeRC && 0 == sysTimeRC) {
+		J9JFRCPULoad *jfrEvent = (J9JFRCPULoad *)reserveBuffer(currentThread, sizeof(J9JFRCPULoad));
+		if (NULL != jfrEvent) {
+			initializeEventFields(currentThread, (J9JFREvent *)jfrEvent, J9JFR_EVENT_TYPE_CPU_LOAD);
+			UDATA numberOfCpus = j9sysinfo_get_number_CPUs_by_type(J9PORT_CPU_ONLINE);
+			U_64 currentTime = (U_64)j9time_nano_time();
+
+			omrthread_process_time_t *prevProcCPUTime = &currentThread->javaVM->jfrState.prevProcCPUTime;
+			I_64 *prevProcTimestamp = &currentThread->javaVM->jfrState.prevProcTimestamp;
+			if (-1 == *prevProcTimestamp) {
+				jfrEvent->jvmUser = 0;
+				jfrEvent->jvmSystem = 0;
+			} else {
+				U_64 timeDelta = currentTime - *prevProcTimestamp;
+				jfrEvent->jvmUser = OMR_MIN((currentProcCPUTime._userTime - prevProcCPUTime->_userTime) / ((double)numberOfCpus * timeDelta), 1.0);
+				jfrEvent->jvmSystem = OMR_MIN((currentProcCPUTime._systemTime - prevProcCPUTime->_systemTime) / ((double)numberOfCpus * timeDelta), 1.0);
+			}
+			*prevProcCPUTime = currentProcCPUTime;
+			*prevProcTimestamp = currentTime;
+
+			J9SysinfoCPUTime *prevSysCPUTime = &currentThread->javaVM->jfrState.prevSysCPUTime;
+			if (-1 == prevSysCPUTime->timestamp) {
+				jfrEvent->machineTotal = 0;
+			} else {
+				jfrEvent->machineTotal = OMR_MIN((currentSysCPUTime.cpuTime - prevSysCPUTime->cpuTime) / ((double)numberOfCpus * (currentSysCPUTime.timestamp - prevSysCPUTime->timestamp)), 1.0);
+			}
+			*prevSysCPUTime = currentSysCPUTime;
+		}
+	}
+
+}
+
+void
+jfrThreadCPULoad(J9VMThread *currentThread, J9VMThread *sampleThread)
+{
+	PORT_ACCESS_FROM_VMC(currentThread);
+	OMRPORT_ACCESS_FROM_J9PORT(PORTLIB);
+
+	omrthread_thread_time_t threadTime;
+
+	int32_t rc = omrthread_get_self_thread_time(&threadTime);
+
+	if (-1 != rc) {
+		I_64 sysTime = (I_64)threadTime.sysTime;
+		I_64 userTime = (I_64)threadTime.userTime;
```

Comment:
The code here should use the same types as in OMR (`int64_t`) making it clear the casts are not required.

Reply:
fixed